Vagrantfile

```diff
@@ -17,39 +39,38 @@ set -e
 
 echo "${PREFIX} Installing pre-reqs..."
 
-# For installing Java 8
-add-apt-repository ppa:webupd8team/java
-
 # For Mesos
 apt-key adv --keyserver keyserver.ubuntu.com --recv E56151BF
 DISTRO=$(lsb_release -is | tr '[:upper:]' '[:lower:]')
 CODENAME=$(lsb_release -cs)
 echo "deb http://repos.mesosphere.io/${DISTRO} ${CODENAME} main" | sudo tee /etc/apt/sources.list.d/mesosphere.list
 
+# Update available packages
 apt-get -o Acquire::http::Timeout=1 -o Acquire::ftp::Timeout=1 -y update
-echo oracle-java8-installer shared/accepted-oracle-license-v1-1 select true | sudo /usr/bin/debconf-set-selections
 
-echo "Installing oracle-java8-installer..."
-install_package "oracle-java8-installer"
-echo "Installing oracle-java8-set-default..."
-apt-get -o Acquire::http::Timeout=1 -o Acquire::ftp::Timeout=1 -y install oracle-java8-set-default
-echo "Installing libcurl3..."
-apt-get -o Acquire::http::Timeout=1 -o Acquire::ftp::Timeout=1 -y install libcurl3
-echo "Installing zookeeperd..."
-apt-get -o Acquire::http::Timeout=1 -o Acquire::ftp::Timeout=1 -y install zookeeperd
-echo "Installing aria2..."
-apt-get -o Acquire::http::Timeout=1 -o Acquire::ftp::Timeout=1 -y install aria2
+# Ensure we have java 8
+java_version=$(java -version 2>&1 | awk -F '"' '/version/ {print $2}')
+if [[ ! "$java_version" > "1.8" ]]; then
+  # For installing Java 8
+  add-apt-repository ppa:webupd8team/java
+
+  echo oracle-java8-installer shared/accepted-oracle-license-v1-1 select true | sudo /usr/bin/debconf-set-selections
 
-echo "${PREFIX}Installing mesos ..."
-apt-get -o Acquire::http::Timeout=1 -o Acquire::ftp::Timeout=1 -y install mesos=0.25.0-0.2.70.ubuntu1404
-echo "Done"
+  install_package "oracle-java8-installer"
+  install_package "oracle-java8-set-default"
+  ln -sf /usr/lib/jvm/java-8-oracle/jre/lib/amd64/server/libjvm.so /usr/lib/libjvm.so
```

Comment:
Just FYI, the symlink creation already existed, I'm just moving it into the "java installation" section.  However, as for "why" we need it -- seems it *might* be needed for `java`.  When I do lsof on the file it reveals java processes that reference it:

```
vagrant@master:~$ sudo lsof /usr/lib/libjvm.so
COMMAND   PID USER  FD   TYPE DEVICE SIZE/OFF   NODE NAME
java    16484 root mem    REG    8,1 16938578 920960 /usr/lib/jvm/java-8-oracle/jre/lib/amd64/server/libjvm.so
java    16485 root mem    REG    8,1 16938578 920960 /usr/lib/jvm/java-8-oracle/jre/lib/amd64/server/libjvm.so
```